src/app/habits/createhabit.tsx

```diff
@@ -0,0 +1,238 @@
+import { icons } from "./icons";
+
+export default function Createhabit({ habitid }: { habitid: String }) {
+  const { colorScheme } = useColorScheme();
+  const borderColor = DefaultColors[colorScheme].tint;
+  const { icon } = useContext(IconContext);
+  const [habitName, setHabitName] = useState("");
+  const [description, setDescription] = useState("");
+  const [color, setColor] = useState("red");
+  const [goalType, setGoalType] = useState("Daily");
+  const [completion, setCompletion] = useState(1);
+  const [failedRequirements, setFailedRequirements] = useState(false);
+
+  const needsTag = icon === "default";
+  const needsName = habitName === "";
+  const canCreateHabit = !needsTag && !needsName;
+
+  useEffect(() => {
+    if (habitid) {
+      // Update icon, habitName, description, color, goalType, and completion
+      // after querying for habit using habitid
+    }
+  }, []);
+
+  return (
+    <View className="flex flex-1 flex-col gap-y-5 px-5 pt-11">
+      {/* Header */}
+      <View className="flex flex-row items-center justify-between">
+        <TouchableOpacity
+          className="flex w-28 flex-row items-center justify-center rounded-2xl border px-2 py-1"
+          style={{ borderColor }}
+          onPress={() => {
+            router.back();
+          }}
+        >
+          <Icon icon={IconX} size={20} />
+          <Text className="ml-1 text-base font-semibold">Cancel</Text>
+        </TouchableOpacity>
+        <Text className="text-base font-semibold">
+          {habitid ? "Edit" : "New"} habit
+        </Text>
+        <TouchableOpacity
+          className="flex w-28 flex-row items-center justify-center rounded-2xl border px-2 py-1"
+          style={{ borderColor }}
+          onPress={() => {
+            if (canCreateHabit) {
+              setFailedRequirements(false);
+              alert("create/edit habit");
+            } else {
+              setFailedRequirements(true);
+              alert(
+                `can't create habit:\n${needsTag ? " - missing tag\n" : ""}${needsName ? " - missing name" : ""}`,
+              );
+            }
+          }}
+        >
+          <Icon icon={habitid ? IconCheck : IconArrowForwardUp} size={20} />
+          <Text className="ml-1 text-base font-semibold">
+            {habitid ? "Done" : "Next"}
+          </Text>
+        </TouchableOpacity>
+      </View>
+
+      {/* Icon & Name */}
+      <View className="flex flex-col gap-y-2">
+        <Text className="text-base font-semibold">Icon & Name</Text>
+        <View className="flex flex-row">
+          <IconButton
+            icon={icons[icon]}
+            onPress={() => {
+              router.push("/habits/habiticons");
+            }}
+            required={needsTag && failedRequirements}
+          />
+          <TextInput
+            className="ml-2 h-10 flex-1 rounded-lg border px-3"
+            numberOfLines={1}
+            style={{
+              borderColor,
```

Comment:
I think instead of having an alert and then red fill, you should grey out the next button when they haven't chosen an icon & title. And if they do press the next button while it's greyed out, show red borders on the icon & description (instead of red fill).